typing/uniqueness_analysis.ml

```diff
@@ -264,14 +284,13 @@ end = struct
 
        and as a result, we can delay the projection at `x`.
 
-       The downside of relaxing is the loss of completeness: if we relax too
-       much the program will fail type check. In the extreme case we relax it to
-       Error which fails type check outright (and extremely sound, hehe).
+      On the other hand, the analysis shouldn't infer a [Pexp_ident] to have a
+      too strong usage, as that might be unsound.
 
-     - The purpose of this uniqueness analysis is to figure out the most relaxed
-     mode for each use, such that we get the best performance, while still
-     type-check. Currently there are really only two choices worth figuring out,
-     Namely
+     - Therefore, the analysis should try to figure out the strongest/lowest
+     usage of each [Pexp_ident], such that we get the best performance, while
+     still being sound. Currently there are really only two choices worth
+     figuring out, namely:
      - borrowed or aliased?
      - aliased or unique?
 
```

Comment:
Another comment I wrote over in #3157 , which you might be interested in:

```
(** Usage algebra
    In this file we track the usage of variables and tags throughout a source file.
    Information can be composed using three operators:
     - seq: sequential composition such as 'foo; bar'
     - par: parallel composition such as '(foo, bar)'
         where evaluation order is not specified.
     - choose: non-deterministic choice such as 'if b then foo else bar'
    subject to the following laws:
     - seq, par, choice are associative
     - par, choice are commutative
     - seq and par both distribute over choice
     - choice is idempotent
     - seq, par, choice have a common unit 'empty'
    Note: These laws do not apply regarding the concrete error messages reported
    to the user if the analysis fails. However, the laws do determine whether
    the analysis may fail in the first place.
    Aside: This structure is also known as a United Monoid (Mokhov et al.)
    It differs from, say, Kleene algebra in that seq and choice share a unit.
    To make this a Kleene algebra we would need a unit for choice that
    annihilates seq. In most presentations, this corresponds to a code path
    that can never be taken like 'assert false'. But we want to perform
    the uniqueness analysis even such branches, since it would be a bad
    user experience if a wrong use of uniqueness could lead to a segfault
    before reporting the failed assertion. *)
```

I include it here in case it inspires you and not asking you to include it -- feel free to resolve this as soon as you read it. In fact, I would prefer we just merge #3157 instead of adopting the comment now.

Reply:
Thanks - I will adopt that.

> In fact, I would prefer we just merge https://github.com/ocaml-flambda/flambda-backend/pull/3157 instead of adopting the comment now.

I feel it's better to have a separate PR to improve the comments that are unrelated to overwriting.

Reply:
Okay!